### PR TITLE
Add time limit to simple maze

### PR DESCRIPTION
**Pull request by mohsen-ghaffari1992**
Simple maze agent has now a limited time frame, which allows us to run both learning and evaluation for many episodes.
Agent will die (without obtaining bad reward) if times out. In this way, we do not need to be worried about the learning policy. because, we only cut the process not add anything to learning model.
This was my suggestion for adding time frame option to the learning process in general that if one wants to set a limitation of running episodes, they can do it without adding more components to the state (you disagree).

### CODE REVIEW

File: src/main/scala/symsim/examples/concrete/simplemaze/Maze.scala [hunk 1]
```diff
@@ -52,16 +52,17 @@ object Maze
     Episodic:
 
   val TimeHorizon: Int = 2000
+  val TimeLimit: Int = 10
 
   def isFinal (s: MazeState): Boolean =
-    s == (4, 3) || s == (4, 2)
+    (s._1, s._2) == (4, 3) || (s._1, s._2) == (4, 2) || s._3 == 100
```
Comment (wasowski) [inline src/main/scala/symsim/examples/concrete/simplemaze/Maze.scala:58]: why this change? Is something weird happening?

Comment (wasowski) [inline src/main/scala/symsim/examples/concrete/simplemaze/Maze.scala:58]: You could test on timeout here.

File: src/main/scala/symsim/examples/concrete/simplemaze/Maze.scala [hunk 2]
```diff
@@ -93,9 +94,12 @@ object Maze
       newState = successor (s) (action)
     yield (newState, mazeReward (newState))
 
-  def initialize: Randomized[MazeState] =
-    Randomized.repeat (Randomized.oneOf (instances.allObservableStates*))
-      .filter (s => !isFinal (s))
+  def initialize: Randomized[MazeState] = for
+    x <- Randomized.repeat(Randomized.between(1, 4))
```
Comment (wasowski) [inline src/main/scala/symsim/examples/concrete/simplemaze/Maze.scala:98]: This might introduce the same initialization problem that I suspect we have in the braking car.

**Comment by wasowski** [comment]
This looks alright - modulo two minor comments.  I am gonna look at PR #226 as if that one works one of my comments falls.

File: src/main/scala/symsim/examples/concrete/simplemaze/Maze.scala (hunk 2 above)
Comment (wasowski) [reply src/main/scala/symsim/examples/concrete/simplemaze/Maze.scala:98]: OK This is not a problem, as analyzed here: https://github.com/itu-square/symsim/pull/226#issuecomment-1870625502

**Comment by wasowski** [comment]
@mohsen-ghaffari1992 I checked the test, and its fine. So only the `isFinal` change and we can merge (while we do not need any changes to eval - but I do not need that PR anywhere, so I suppose you did not even push).

**Comment by wasowski** [comment]
> Simple maze agent has now a limited time frame, which allows us to run both learning and evaluation for many episodes. Agent will die (without obtaining bad reward) if times out. In this way, we do not need to be worried about the learning policy. because, we only cut the process not add anything to learning model. This was my suggestion for adding time frame option to the learning process in general that if one wants to set a limitation of running episodes, they can do it without adding more components to the state (you disagree).

This is actually a very good sugestion and I agree with it (except that I think we should do everything, also the cutoff here, and do not touch symsim code at all).

**Comment by mohsen-ghaffari1992** [comment]
@wasowski 
PR is ready to merge if you have no more comments.
